### ISSUE TITLE
Configurations for PRO-cap benchmarking

### ISSUE THREAD

**Issue by MasayukiNagai** [jmschrei/cherimoya#36]
Thanks for sharing Cherimoya and the detailed documentation!

I’m interested in benchmarking Cherimoya on PRO-cap data against ProCapNet and other models. I saw in the documentation that the default architecture and training settings were already selected through large-scale hyperparameter exploration, so I’m planning to start with those defaults.

Have you already tried Cherimoya on PRO-cap or other transcription-initiation data? If so, are there any changes you would recommend from the default configuration? Or would you expect the default configuration to be a reasonable starting point without an additional parameter sweep?

Thanks for any guidance!

**Comment by adamyhe** [comment]
Hi Moon,

I've been working on training ProCapNet/BPNet and Cherimoya models for the ~200 recent PRO-cap experiments from ENCODE. I've found default settings work reasonably well (128 filters). I'm still tracking down some issues with the choice of epochs (20 max, 2 warmup seems to be better on counts than 50 max, 5 warmup, but worse on profile), but I should hopefully have those done soon-ish?

Anyway happy to share the ProCapNet/BPNet models if you'd like to test those out in the meantime, or to collaborate on Cherimoya/PRO-cap if you'd like. :)

Adam

**Comment by MasayukiNagai** [comment]
Hi Adam,

Thanks for the response! It’s very cool that you’re working on benchmarking such a large set of PRO-cap experiments.

I was wondering whether you changed the positive/negative ratio from the default settings. ProCapNet uses source_fracs: [0.875, 0.125], while Cherimoya uses negative_ratio: 0.25. I ran two versions of Cherimoya with negative ratios of 0.125 and 0.25, and the choice made a pretty large difference in predictive performance. So I was curious what ratio you’ve been using for your experiments.

Thanks!
Moon

**Comment by adamyhe** [comment]
I've used 0.125 for the BPNet architecture and 0.25 for Cherimoya. I haven't actually tested 0.125 on Cherimoya. Are you seeing that 0.125 is much worse or better?

**Comment by MasayukiNagai** [comment]
When I used 0.25 for Cherimoya, its performance was even lower than that of ProCapNet. However, reducing the value to 0.125 improved Cherimoya’s performance. This is both on counts and profiles but only tested in K562 fold1. Overall:

0.125 Cherimoya > 0.125 ProCapNet > 0.25 Cherimoya

Is this consistent with what you've seen? Or I might've misconfigured something.

**Comment by adamyhe** [comment]
[bpnet_vs_cherimoya_profile_jsd.pdf](https://github.com/user-attachments/files/30908554/bpnet_vs_cherimoya_profile_jsd.pdf)
[bpnet_vs_cherimoya_log_counts_pearson.pdf](https://github.com/user-attachments/files/30908555/bpnet_vs_cherimoya_log_counts_pearson.pdf)

This is with 0.25, max_epochs=20, early_stopping=5, warmup=2, across all 224 ENCODE PRO-cap experiments.

I will note that I've 1) dropped the mappability mask on the profile loss for both architectures (my BPNet/orig ProCapNet models are trained using vanilla bpnet-lite v1.0.0, and Cherimoya with its default loss functions/weighting), 2) switched to using GC-matched negatives instead of untranscribed DHS, and 3) switched to using plain JSD instead of the normalized JSD score they used in the original ProCapNet paper.

**Comment by jmschrei** [comment]
Using n_filters=192, batch_size=64, max_epochs=50, early_stopping=50 on the exact same data and loci as the original ProCapNet models, and compared against the released models on their validation set, I get this:

<img width="1431" height="517" alt="Image" src="https://github.com/user-attachments/assets/49367a8f-62a8-498c-a1ff-e74095771ae1" />

The improvement in log count prediction seems in line with what @adamyhe is seeing.

**Comment by MasayukiNagai** [comment]
Thanks both for sharing the results! 

@jmschrei By "the same data", do you also mean the same positive/negative (DHS untrascribed) split? Or were the negatives generated using the GC-matched negative sampling from the default Cherimoya pipeline? 

Also, does each point in the figure correspond to a cell line? In the ProCapNet paper, I only see six cell lines: K562, A673, Caco-2, Calu3, HUVEC, and MCF10A.

**Comment by jmschrei** [comment]
I'm using the exact training and test regions used in the ProCapNet paper (including their method for calling peaks and negatives), which are released alongside the models: https://www.encodeproject.org/annotations/ENCSR801ECP/

This is all seven folds for each cell line.

**Comment by MasayukiNagai** [comment]
Great, that makes sense. Thank you so much!
